src/lib/sticky-header/sticky-header.ts

```diff
@@ -0,0 +1,299 @@
+
+/**
+ * Directive that marks an element as a sticky-header. Inside of a scrolling container (marked with
+ * cdkScrollable), this header will "stick" to the top of the scrolling viewport while its sticky
+ * region (see cdkStickyRegion) is in view.
+ */
+@Directive({
+  selector: '[cdkStickyHeader]',
+})
+export class CdkStickyHeader implements OnDestroy, AfterViewInit {
+
+  /** z-index to be applied to the sticky header (default is 10). */
+  @Input('cdkStickyHeaderZIndex') zIndex: number = 10;
+
+  /** boolean value to mark whether the current header is stuck*/
+  isStuck: boolean = false;
+  /** Whether the browser support CSS sticky positioning. */
+  private _isPositionStickySupported: boolean = true;
+
+  /** The element with the 'cdkStickyHeader' tag. */
+  element: HTMLElement;
+  /** The upper container element with the 'cdkStickyRegion' tag. */
+  stickyParent: HTMLElement | null;
+  /** The upper scrollable container. */
+  upperScrollableContainer: HTMLElement;
+  /**
+   * The original css of the sticky element, used to reset the sticky element
+   * when it is being unstuck
+   */
+  private _originalStyles = {} as CSSStyleDeclaration;
+  /**
+   * 'getBoundingClientRect().top' of CdkStickyRegion of current sticky header.
+   * It is used with '_stickyRegionBottomThreshold' to judge whether the current header
+   * need to be stuck.
+   */
+  private _stickyRegionTop: number;
+  /**
+   * Bottom of the sticky region offset by the height of the sticky header.
+   * Once the sticky header is scrolled to this position it will stay in place
+   * so that it will scroll naturally out of view with the rest of the sticky region.
+   */
+  private _stickyRegionBottomThreshold: number;
+
+  private _onScrollSubscription: Subscription;
+
+  private _onTouchSubscription: Subscription;
+
+  private _onResizeSubscription: Subscription;
+
+  constructor(element: ElementRef,
+              scrollable: Scrollable,
+              @Optional() public parentRegion: CdkStickyRegion,
+              platform: Platform) {
+    if (platform.isBrowser) {
+      this.element = element.nativeElement;
+      this.upperScrollableContainer = scrollable.getElementRef().nativeElement;
+      this._setStrategyAccordingToCompatibility();
+    }
+  }
+
+  ngAfterViewInit(): void {
+    if (!this._isPositionStickySupported) {
+
+      this.stickyParent = this.parentRegion != null ?
+        this.parentRegion._elementRef.nativeElement : this.element.parentElement;
+
+      let values = window.getComputedStyle(this.element, '');
+      this._originalStyles = {
+        position: values.position,
+        top: values.top,
+        right: values.right,
+        left: values.left,
+        bottom: values.bottom,
+        width: values.width,
+        zIndex: values.zIndex} as CSSStyleDeclaration;
+
+      this.attach();
+      this.defineRestrictionsAndStick();
+    }
+  }
+
+  ngOnDestroy(): void {
+    if (this._onScrollSubscription) {
```

Comment:
:+1:  Changed it to  `'[this._onScrollSubscription, this._onScrollSubscription, this._onResizeSubscription]
        .forEach(s => s && s.unsubscribe());'`